message_set.go

```diff
@@ -48,9 +48,11 @@ func (msb *MessageBlock) decode(pd packetDecoder) (err error) {
 type MessageSet struct {
 	PartialTrailingMessage bool // whether the set on the wire contained an incomplete trailing MessageBlock
 	Messages               []*MessageBlock
+	UnderlyingMessageCount int // in case of compressed messages wraps in a single fake "message"
```

Comment:
I believe that once N compressed messages are stored inside a single fake "message" we lose the amount of messages that was serialized and `len(ms.Messages)` would then return 1 instead of N.